### DISCUSSION TITLE
v0.2.0

### DISCUSSION THREAD

**Discussion by carson-katri** [carson-katri/dream-textures#643] [Announcements]
This update brings ControlNet, optimization, and nodes!

Check out the new [render engine](https://github.com/carson-katri/dream-textures/wiki/Render-Engine) to get started with the node system.

Join the [Dream Textures Discord server](https://discord.gg/EmDJ8CaWZ7) to discuss the add-on, share your art, and get help from other users.

## Choose Your Installation
Several versions are available. Find the one that will work for you.

<details>
<summary>Windows</summary>

### NVIDIA GPU
* [Dream Textures for Windows \(CUDA\)](https://github.com/carson-katri/dream-textures/releases/download/0.2.0/dream_textures-windows-cuda.7z) - Local and cloud rendering with NVIDIA CUDA
    * Due to file size limits on GitHub, the add-on ZIP file is contained in a 7-Zip archive.
    * Install [7-Zip](https://www.7-zip.org/), right click on the downloaded `dream_textures-windows-cuda.7z` archive, select "7-Zip" > "Extract Here" to get the `dream_textures-windows-cuda.zip` file.
    * In Blender preferences open "Add-ons" > "Install...", then choose the `dream_textures-windows-cuda.zip` file.

### AMD GPU
* [Dream Textures for Windows \(DirectML\)](https://github.com/carson-katri/dream-textures/releases/download/0.2.0/dream_textures-windows-directml.zip) - Local and cloud rendering with DirectML
</details>

<details>
<summary>macOS</summary>

### Apple Silicon (M1/M2)
* [Dream Textures for Apple Silicon](https://github.com/carson-katri/dream-textures/releases/download/0.2.0/dream_textures-macos-arm.zip) - Local and cloud rendering for Apple Silicon Macs

### Intel
Dream Textures is not currently available for Intel Macs.
</details>

<details>
<summary>Linux</summary>

Installation must be completed manually. Follow the instructions to [install manually](https://github.com/carson-katri/dream-textures/blob/main/docs/DEVELOPMENT_ENVIRONMENT.md) for more details.
</details>

<details>
<summary>
<a href="https://www.blendermarket.com/products/dream-textures">Blender Market</a>
</summary>

You can optionally purchase it for a small fee on Blender Market.

> The version on Blender Market is identical to the free download available here. It is simply a way to support my work on this tool.

</details>

> **Note** DreamStudio builds will no longer be released. Please see the official [Stability for Blender](https://platform.stability.ai/docs/integrations/blender) addon for DreamStudio integration.

## Setup
After installing the appropriate add-on ZIP file, enable the add-on and expand its preferences. There you will find further instructions for setup.

See the [setup guide](https://github.com/carson-katri/dream-textures/blob/main/docs/SETUP.md) for more detailed instructions.

### ControlNet
ControlNet gives you a lot more control over image generation. It has been fully integrated into the new [render engine](https://github.com/carson-katri/dream-textures/wiki/Render-Engine), as well as in the Image Editor and [texture projection](https://github.com/carson-katri/dream-textures/wiki/Texture-Projection) panel.

Here are some things you can do with it.

1. Turn a pose into an image

https://user-images.githubusercontent.com/13581484/233731787-a07b13d3-c5f2-48d4-86ea-ba39279fd504.mov

2. Colorize a Grease Pencil sketch

https://user-images.githubusercontent.com/13581484/233732050-102a6473-874e-4f6b-9a4b-dccf35316422.mov

3. Render a scene from a basic mesh

https://user-images.githubusercontent.com/13581484/233732104-84a6f3b7-5d2d-4544-aa0b-5ca67b1eddd4.mov

4. Improved texture projection fidelity, and image editing with ip2p

https://user-images.githubusercontent.com/13581484/233732182-c688fedc-496d-41ca-84c7-d2f1408d6676.mov

### Optimizations
PyTorch 2.0 is now bundled with the release. This brings the new SDP attention which reduces memory and generates faster. This new optimization is enabled by default.

A new CFG End optimization is also included which can be used to disabled classifier-free guidance after a certain percentage of steps. This can be great for quick prototyping.

The optimization UI is also improved to provide a better overview of what each option does. The time each generation takes is now displayed below the "Generate" button.

### Render Engine
A dedicated "Dream Textures" render engine is now available. It provides the node system and acts as an alternative to engines like Cycles or Eevee.

Learn more about the render engine and node system [in the wiki](https://github.com/carson-katri/dream-textures/wiki/Render-Engine).

## Full Changelog
* Add ControlNet by @carson-katri in https://github.com/carson-katri/dream-textures/pull/585
* Optimization Fixes and Improvements by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/575
* Add segmentation ControlNet support by @carson-katri in https://github.com/carson-katri/dream-textures/pull/614
* Support Inpainting with ControlNet by @carson-katri in https://github.com/carson-katri/dream-textures/pull/625
* Add CFG End Optimization by @carson-katri in https://github.com/carson-katri/dream-textures/pull/617
* Bump Torch ROCm version to 2.0 by @cornpo in https://github.com/carson-katri/dream-textures/pull/636
* Add normal map generation by @carson-katri in https://github.com/carson-katri/dream-textures/pull/632
* Release v0.2.0 by @carson-katri in https://github.com/carson-katri/dream-textures/pull/638

## New Contributors
* @cornpo made their first contribution in https://github.com/carson-katri/dream-textures/pull/636

**Full Changelog**: https://github.com/carson-katri/dream-textures/compare/0.1.0...0.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/carson-katri/dream-textures/releases/tag/0.2.0'>v0.2.0</a>.</em>

**Comment by benrugg** [comment]
This looks great! ControlNet is such an amazing improvement for some of these workflows. Congrats on this release, @carson-katri!

**Reply by carson-katri** [reply]
Thanks!

**Comment by jjp9624022** [comment]
lora please
